lara-typescript/src/section-authoring/api/util/finding-utils.ts

```diff
@@ -0,0 +1,54 @@
+import { IPage, SectionId, ISection, PageId } from "../api-types";
+
+export interface ISectionAddress {
+  pageIndex: number|null;
+  sectionIndex: number|null;
+}
+
+// memoize me, invalidate on pages[]
+export const findPage = (pages: IPage[], pageId: PageId): IPage|null => {
+  return pages.find(p => p.id === pageId) || null;
+};
+
+// memoize me, invalidate on pages[]
+export const findSection = (pages: IPage[], sectionId: SectionId): ISection|null => {
+  for (const page of pages) {
+    for (const section of page.sections) {
+      if (section.id === sectionId) {
+        return section;
+      }
+    }
+  }
+  return null;
+};
+
+export const findSectionByAddress = (pages: IPage[], address: ISectionAddress) => {
+  const { pageIndex, sectionIndex } = address;
+  if (pageIndex !== null && sectionIndex !== null) {
```

Comment:
```suggestion
  if (pageIndex != null && sectionIndex != null) {
```
I generally prefer this form because it handles `undefined` as well, which seems like it should be handled similarly if it were ever to occur.